building/upstream-check/snapshot-pull.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+import subprocess
+import apt_pkg
+import sys
+import os
+apt_pkg.init_system()
+
+if len(sys.argv) != 2 or sys.argv[1][0] != "2" or sys.argv[1][-1] != "Z":
+	print("usage: snapshot-pull.py <snapshot-version>")
+	print("   version example: 20180710T043017Z")
+	sys.exit(1)
+
+VERSION = sys.argv[1]
+prefix = "snapshot.debian.org/archive/debian/%s" % VERSION
+
+def download(path):
```

Comment:
Why not use an existing solution to mirror the repository? Also, since apt_pkg is a dependency anyway, why not use it to parse the package files?

Reply:
I can't find good documentation for apt_pkg, and it doesn't look like it would be straightforward to integrate. I'd rather stick with this code for now and leave migration for later.